[jit] fix indexing into a tensor with a tuple

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #79335

As title.

Differential Revision: [D37089015](https://our.internmc.facebook.com/intern/diff/D37089015)

suo committed:
[jit] fix indexing into a tensor with a tuple

As title.

[ghstack-poisoned]

suo commented:
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37089015).

suo committed:
Update on "[jit] fix indexing into a tensor with a tuple"


As title.

Differential Revision: [D37089015](https://our.internmc.facebook.com/intern/diff/D37089015)

[ghstack-poisoned]

suo committed:
Update on "[jit] fix indexing into a tensor with a tuple"


As title.

Differential Revision: [D37089015](https://our.internmc.facebook.com/intern/diff/D37089015)

[ghstack-poisoned]

suo committed:
Update on "[jit] fix indexing into a tensor with a tuple"


As title.

Differential Revision: [D37089015](https://our.internmc.facebook.com/intern/diff/D37089015)

[ghstack-poisoned]

suo committed:
Update on "[jit] fix indexing into a tensor with a tuple"


As title.

Differential Revision: [D37089015](https://our.internmc.facebook.com/intern/diff/D37089015)

[ghstack-poisoned]

suo commented:
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37089015).

gmagogsfm commented on torch/csrc/jit/frontend/ir_emitter.cpp:
```diff
@@ -3639,8 +3639,12 @@ struct to_ir {
+        if (input->type()->kind() == TypeKind::TupleType) {
+          return std::make_shared<SimpleValue>(
+              emitIndex(loc, self, createTupleUnpack(input)));
+        } else {
```
Nit, early return makes `else` unnecessary.
if (xx):
  return yy

return zz

suo committed:
Update on "[jit] fix indexing into a tensor with a tuple"


As title.

Differential Revision: [D37089015](https://our.internmc.facebook.com/intern/diff/D37089015)

[ghstack-poisoned]

suo commented:
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37089015).
